modules/apps/frontend-editor/frontend-editor-ckeditor-web/src/main/resources/META-INF/resources/ckeditor.jsp

```diff
@@ -624,10 +624,15 @@ name = HtmlUtil.escapeJS(name);
 
 		ckEditor.on('drop', function (event) {
 			var data = event.data.dataTransfer.getData('text/html');
-			var fragment = CKEDITOR.htmlParser.fragment.fromHtml(data);
-			var name = fragment.children[0].name;
-			if (name) {
-				return this.pasteFilter.check(name);
+
+			if (data) {
+				var fragment = CKEDITOR.htmlParser.fragment.fromHtml(data);
+
+				var name = fragment.children[0].name;
+
+				if (name) {
+					return this.pasteFilter.check(name);
```

Comment:
What was this doing? I don't remember 😂

Reply:
The change in this PR is very much needed, as the current code breaks on image drop:
>Cannot read property 'name' of undefined

The change is a simple `if`, making us parse HTML only if data is HTML type. But, I am not sure if that code now does what it is suppose to, or if it did before. The [commit message](https://github.com/antonio-ortega/liferay-portal/pull/117/commits/0f26f6b108ec430c8a1fdccc8e3d432199643fb5) is 
> Prevent dropping images on description editor

The code before this PR does technically prevent dropping images, but does it by breaking JS and throwing and error in log. @julien could you please take a look, there might be something I'm missing here.

In any case, if this is a bug, I don't think it is a blocker for this PR, we can do it as a separate task.

Reply:
Hey @markocikos, I was simply referring to **that** specific line:

`this.pasteFilter.check(name)`

I was just curious as to what were we doing there... I would assume aborting the drop if the thing was not in the `allowedContent` config... but can't tell for sure what is what.

Reply:
If I understand correctly, that's a way to use [`pasteFilter` feature](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_config.html#cfg-pasteFilter) that we have added to config [here](https://github.com/liferay/liferay-portal/blob/master/modules/apps/journal/journal-web/src/main/java/com/liferay/journal/web/internal/editor/configuration/JournalArticleDescriptionEditorConfigContributor.java#L54).

Reply:
If we want to apply some HTML restrictions, this seems fine. But, if we are using this to prevent image drops, it may not be the best approach.

Reply:
Okay, maybe as we advance in this feature we'll have a better idea of what we need to do